scripts/tasks/ts.js

```diff
@@ -10,6 +10,6 @@ module.exports = function (options) {
 
   if (options.isProduction) {
     execSync(typescriptPath + ' -outDir lib-amd -t es5 -m amd ' + extraParams);
-    // execSync(typescriptPath + ' -outDir lib-es6 -t es6 -m es6 ' + extraParams, 'typescript es6');
+    execSync(typescriptPath + ' -outDir lib-es2015 -t es2015 -m es2015 ' + extraParams);
```

Comment:
Is there a difference between es6 and es2015? I'm trying to understand why you changed this to es2015 from es6.

Aso, is es2015/es6 correct for target? That means the consumer would need to use babel for transpiling into browser happy code. maybe es5 is the right target here (but es6 or esnext for module)

Also in doing this, have you had any luck consuming the modules from rollup and seeing improvement?

I had a separate PR (which I closed temporarily) to turn this on. I did not check it in because in experimenting with tree shaking in webpack, it wouldn't work correctly (yet.) I've been working with the webpack folks on v4, which adds support for tagging packages to have no side-effects which enables global code (like the loadStyles function calls) to only be included if other exports in the file are imported.
 because there is code executed in the global scope (loadStyles calls on sass files) which basically brings all the things in anyway. Frustrating. But Webpack 4 has a feature where you can specify to only include sideEffect (global) code if something in the file is explicit imported.

I'm happy to add this but I want to make sure we're not doing this without a reason. Is es2015 the right target/module format?

Reply:
@dzearing So I just compared the emitted output for the Label Component from `-m es2015` and `-m es6` and found no difference. I went with `es2015` here simple because that's the "more official" name for that year of changes to the spec, but I'm more than happy to change it back to `es6` if that's what you would prefer.

I think you're correct that `es2015` isn't the correct target, I assumed you had to target `es2015` if your set the module to `es2015` but, looking at the TypeScript Compiler Options, you can still target `es5`, I'll go ahead and push a change to that.

And this was less motivated by making sure the tree-shaking worked properly and that there was improvement in the bundle size, and more about just the minimum bar of being able to bundle with Rollup, without having to put every Component in the `namedExports` of the CommonJS plugin.

I have verified in a test project, that I can bundle Fabric Components, with the es2015 modules, without having to include the commonjs plugin, or put them in the `namedExports` array. I'm happy to send you that project if you'd like to take a look.

Comparing the two bundles with Rollup (es2015 vs. CommonJS), that are just importing one Component, it looks to save ~1KB but that's probably negligible due to the extra code added by the Rollup CommonJS plugin to unwrap the Components. But like I said, I'm more looking to just be able to bundle Fabric Components with Rollup, without having to list every Component in the `namedExports`.

Reply:
Thanks a lot Kyle for this, good to have help in this area! :D The savings will be variable, depending on the scenario and imports used. A good test is probably bundling something simple like this:

index.ts
```
export { getWindow } from '@uifabric/utilities';
```

Then, working up to:

```
export { getWindow } from 'office-ui-fabric-react/lib/Utilities';
```

and then the ideal, which is what I want to get to:
```
export { getWindow } from 'office-ui-fabric-react';
```
Where each should only export the `getWindow` function.